yarn-project/types/src/l2_block.ts

```diff
@@ -321,9 +321,10 @@ export class L2Block {
   /**
    * Deserializes L2 block without logs from a buffer.
    * @param buf - A serialized L2 block.
+   * @param blockHash - The hash of the block.
    * @returns Deserialized L2 block.
    */
-  static fromBuffer(buf: Buffer | BufferReader) {
+  static fromBuffer(buf: Buffer | BufferReader, blockHash?: Buffer) {
```

Comment:
Judging from this change, wouldn't it be easier to just squeeze `blockHash` as another of the fields of `L2Block`? Or does it break serialization to/from noir?

Reply:
It looks like this is planned as part of #3938 ?
https://github.com/AztecProtocol/aztec-packages/blob/f81b7c050a22e326de3e2009dced856e18c1686f/yarn-project/circuit-types/src/l2_block.ts#L358

Reply:
I can't find it now but there was an issue/PR about blocks now being serialisable because they didn't logs.

I'd consider doing this but only if I make `blockHash` mandatory in the constructor so it doesn't cause any errors downstream 😬